### DISCUSSION TITLE
untrusted user input with x-text?

### DISCUSSION THREAD

**Discussion by benkoshy** [alpinejs/alpine#3071] [1. Help] [answered]
I have some user input that I wish to display, back on the webpage:

```html
<p x-text="`${untrusted_user_input}  `">
</p>
```
[x-text documentation](https://alpinejs.dev/directives/text)

Is the above code safe? Do I need to sanitize before displaying it to the user?

Any pointers would be much appreciated.

**Comment by jimmywarting** [comment, marked as answer]
x-text uses innerText, so it should be safe to use.
you can't have any html tags or attributes when setting the content using innerText.
